src/models/connectionStore.ts

```diff
@@ -448,34 +521,39 @@ export class ConnectionStore {
         type: CredentialsQuickPickItemType,
         isConnectionString: boolean = false,
     ): Promise<boolean> {
-        let self = this;
         let password = isConnectionString ? conn.connectionString : conn.password;
-        return new Promise<boolean>(async (resolve, reject) => {
-            if (Utils.isNotEmpty(password)) {
-                let credType: string =
-                    type === CredentialsQuickPickItemType.Mru
-                        ? ConnectionStore.CRED_MRU_USER
-                        : ConnectionStore.CRED_PROFILE_USER;
-                let credentialId = ConnectionStore.formatCredentialId(
-                    conn.server,
-                    conn.database,
-                    conn.user,
-                    credType,
-                    isConnectionString,
-                );
-                await self._credentialStore
-                    .saveCredential(credentialId, password)
-                    .then((result) => {
-                        resolve(result);
-                    })
-                    .catch((err) => {
-                        // Bubble up error if there was a problem executing the set command
-                        reject(err);
-                    });
-            } else {
-                resolve(true);
+
+        if (Utils.isEmpty(password)) {
+            return true;
+        }
+
+        const profile = conn as IConnectionProfile;
+
+        let legacyCredentialId = ConnectionStore.formatCredentialId(
```

Comment:
Probably should capture some telemetry about whether the old credential key found something so that we can inform a decision about when to drop support for the old one.